components/eventing-controller/api/v1alpha1/subscription_conversion.go

```diff
@@ -136,56 +136,79 @@ func (src *Subscription) setV2ProtocolFields(dst *v1alpha2.Subscription) {
 		}
 		// webhookAuth fields
 		if src.Spec.ProtocolSettings.WebhookAuth != nil {
-			dst.Spec.Config[v1alpha2.WebhookAuthType] = src.Spec.ProtocolSettings.WebhookAuth.Type
+			if src.Spec.ProtocolSettings.WebhookAuth.Type != "" {
```

Comment:
Any particular reason, why we are only checking for empty string for `Type`? Why not also for other fields e.g. `GrantType`, `ClientID` etc.?